jobsdb/jobsdb.go

```diff
@@ -1583,22 +1569,26 @@ func (jd *HandleT) createDSInTx(tx *sql.Tx, newDS dataSetT, l lock.DSListLockTok
 		return err
 	}
 
-	if l != nil {
-		err = jd.setSequenceNumberInTx(tx, l, newDS.Index)
-		if err != nil {
-			return err
-		}
+	err = jd.setSequenceNumberInTx(tx, newDS.Index)
+	if err != nil {
+		return err
 	}
 
 	err = jd.journalMarkDoneInTx(tx, opID)
 	if err != nil {
 		return err
 	}
 
+	// Tracking time interval between new ds creations. Hence calling end before start
+	if jd.isStatNewDSPeriodInitialized {
```

Comment:
previously, we were were not tracking time, if new DS was added because of migration. But, with this change we will always be tracking the time interval between new DS creations. Is that fine?

Reply:
kindly ignore, this is not valid any anymore.